mm2src/lp_ordermatch/simple_market_maker.rs

```diff
@@ -314,8 +274,25 @@ async fn create_single_order(
 
     checks_order_prerequisites(&rates, &cfg, key_trade_pair.clone()).await?;
 
-    let base_balance = coin_find_and_checks(cfg.base.clone(), key_trade_pair.clone(), true, ctx).await?;
-    coin_find_and_checks(cfg.rel.clone(), key_trade_pair.clone(), false, ctx).await?;
+    let base_coin = lp_coinfind(ctx, cfg.base.as_str())
+        .await?
+        .ok_or(MmError::new(OrderProcessingError::AssetError))?;
+
+    let base_balance = match base_coin.get_non_zero_balance() {
+        Ok(x) => MmNumber::from(x.spendable),
+        Err(err) => {
+            return match err.into_inner() {
+                BalanceError::BalanceIsZero => Err(MmError::new(OrderProcessingError::BalanceIsZero)),
+                BalanceError::Transport(_) | BalanceError::InvalidResponse(_) | BalanceError::Internal(_) => {
+                    Err(MmError::new(OrderProcessingError::BalanceInternalError))
+                },
+            }
+        },
+    };
```

Comment:
This is also ugly because it's not possible to do a from `MmError<BalanceError>>` to `MmError<OrderProcessingError>` since BalanceError belongs to another crate.

I would have love todo `let base_balance = MmNumber::from(base_coin.get_non_zero_balance()?.spendable)`